ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -0,0 +1,125 @@
+ * @property {string} recipientAddress    - the Ethereum address of the recipient
+ */
+
+/**
+ * useTransactionDisplayData
+ *
+ * The goal of this method is to perform all of the necessary computation and
+ * state access required to take a transactionGroup and derive from it a shape
+ * of data that can power all views related to a transaction. Presently the main
+ * case is for shared logic between transaction-list-item and transaction-detail-view
+ * @param {Object} transactionGroup group of transactions
+ * @return {TransactionDisplayData}
+ */
+export function useTransactionDisplayData (transactionGroup) {
+  const knownTokens = useSelector(tokenSelector)
+  const t = useI18nContext()
+  const { initialTransaction, primaryTransaction } = transactionGroup
+  // initialTransaction contains the data we need to derive the primary purpose of this transaction group
+  const { transactionCategory } = initialTransaction
+
+  const { from: senderAddress, to } = initialTransaction.txParams || {}
+
+  // for smart contract interactions, methodData can be used to derive the name of the action being taken
+  const methodData = useSelector((state) => getKnownMethodData(state, initialTransaction?.txParams?.data)) || {}
+
+  const actionKey = getTransactionActionKey(initialTransaction)
+  const statusKey = getStatusKey(primaryTransaction)
+  const status = statusKey in pendingStatusMap ? 'pending' : statusKey
+
+  const primaryValue = primaryTransaction.txParams?.value
+  let prefix = '-'
+  const date = formatDateWithYearContext(initialTransaction.time || 0)
+  let subtitle = date
+  let recipientAddress = to
+
+  const token = knownTokens.find((token) => token.address === recipientAddress)
+  const tokenData = useTokenData(initialTransaction?.txParams?.data)
+  const tokenDisplayValue = useTokenDisplayValue(initialTransaction?.txParams?.data, token)
+
+  let category
+  let title
+  // There are four types of transaction entries that are currently differentiated in the design
+  // 1. Send (sendEth sendTokens)
+  // 2. Deposit
+  // 3. Site interaction
+  // 4. Approval
+  if (transactionCategory == null) {
+    const origin = initialTransaction.msgParams?.origin || initialTransaction.origin
+    category = SIGNATURE_REQUEST
+    title = t('signatureRequest')
+    subtitle += origin ? ` · ${origin}` : ''
+  } else if (transactionCategory === 'approve') {
+    category = APPROVAL
+    title = t('approveSpendLimit')
+    subtitle += ` · ${initialTransaction.origin}`
+  } else if (transactionCategory === 'contractDeployment' || transactionCategory === 'contractInteraction') {
+    category = INTERACTION
+    title = (methodData?.name && camelCaseToCapitalize(methodData.name)) || (actionKey && t(actionKey)) || ''
+    subtitle += ` · ${initialTransaction.origin}`
+  } else if (transactionCategory === 'incoming') {
+    category = RECEIVE
+    title = t('receive')
+    prefix = ''
+    subtitle += ` · From: ${shortenAddress(senderAddress)}`
```

Comment:
These `From` and `To` string literals should be using localized strings, with the address as input.

The ` · ` part seems fine as a string-literal though.